File: 4.Sorting/3.InsertionSort/solution.java
public class solution {
 
    public static void insertionSort(int[] arr){
        for(int i = 1; i < arr.length; i++){
            int element = arr[i];
            int j = 0;
            for(j = i-1; j >= 0; j--){
                if(arr[j] > element){
                    arr[j+1] = arr[j];
                }else{
                    break; // arr[j] <= element
                }
            }
            arr[j+1] = element;
        }
    }

    public static void main(String args[]){
        int arr[] = {5,4,3,2,1};
        insertionSort(arr);
        for(int element : arr ){
            System.out.print(element+" ");
        }
    }
}

File: 4.Sorting/3.InsertionSort/solution.js

function insertionSort(nums){

    for(let i = 1; i < nums.length; i++){
        let element = nums[i];
        let j = 0;
        for(j = i-1; j >= 0; j--){
            if(nums[j] > element){
                nums[j+1] = nums[j];
            }
            else{
                break; // arr[j] <= element // Not neccessary
            }
        }
        nums[j+1] = element;
    }
}

const arr = [5,4,3,2,1];
insertionSort(arr);
console.log(arr);
